### PR TITLE
Update filereader-readasdataurl.md

### PR DESCRIPTION
**Pull request by nem035**
@kapnobatai137 we should probably add a footnote (or a glossary item?) for what base 64 encoding is and why it's popular.

### CODE REVIEW

File: javascript/browser-apis/reading-writing/filereader-readasdataurl.md [hunk 1]
```diff
@@ -35,42 +32,52 @@ links:
 ---
 ## Content
 
-The `readAsDataURL()` method is used with file or blob objects. When the read operation is complete it returns a *data URL* with the files contents as base64 encoded string which can be access through `reader.result`.
+The `FileReader` is a global object that provides multuple methods to read the contents of a file stored on the user's computer.
```
Comment (nem035) [inline javascript/browser-apis/reading-writing/filereader-readasdataurl.md:35]:
```suggestion
The `FileReader` is a global object that provides multiple methods to read the contents of a file stored on the user's computer.
```

**Comment by enki-hq** [comment]
### Insight previews:
- [ ] [enkidevs/curriculum/javascript/browser-apis/reading-writing/filereader-readasdataurl.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/browser-apis/reading-writing/filereader-readasdataurl.md?ref=216a15818ea85e7ef4a3d591dd5b0535d1ab9366)

File: javascript/browser-apis/reading-writing/filereader-readasdataurl.md [hunk 2]
```diff
@@ -35,42 +32,52 @@ links:
...
+
+Each method can convert a file into a specific format. 
+
+The `readAsDataURL()` method is used to read the contents of a file and return a *data URL* that represents the contents of a file as a base64 encoded string.
```
Comment (nem035) [inline javascript/browser-apis/reading-writing/filereader-readasdataurl.md:39]:
```suggestion
The `readAsDataURL()` method is used to read the contents of a file and return a *data URL* that represents the contents as a base64 encoded string.
```

File: javascript/browser-apis/reading-writing/filereader-readasdataurl.md [hunk 3]
```diff
@@ -35,42 +32,52 @@ links:
...
 
-This functionality is often used to create thumbnail previews of uploaded image files.
+> The data given to `readDataAsURL` can be a `File` or a `Blob` object. A `File` object is just a specific version of a `Blob` (it can be used anywhere a `Blob` can)
+
+Here's an example of how we can use this to upload an image file from your computer and show it on a website:
```
Comment (nem035) [inline javascript/browser-apis/reading-writing/filereader-readasdataurl.md:43]:
```suggestion
Here's an example of how we can use this to upload an image file from your computer and show it on a website:

> The trick in the code below is that `Image` element source property `src` supports base64 encoded values
```

**Comment by enki-hq** [comment]
Rule insight:footnotes failed for files: 
  - javascript/browser-apis/reading-writing/filereader-readasdataurl.md

**Comment by enki-hq** [comment]
Rule insight:footnotes failed for files: 
  - javascript/browser-apis/reading-writing/filereader-readasdataurl.md

**Comment by kapnobatai137** [comment]
> Rule insight:footnotes failed for files:
> 
>     * javascript/browser-apis/reading-writing/filereader-readasdataurl.md

seems that adding a glossary reference 'breaks' footnotes in this insight.

on line 54 we have `  ).files[0];`, which wasn't triggered as being a footnote because there was no footnote section.

now that we added a glossary reference, I think the `[0]` is considered a footnote

File: glossary/general/base64-encoding.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Base64 Encoding
+
+Base64 is a conversion method used to represent binary data. It does so by creating a "dictionary" where each Base64 digit is represented by exactly 6 bits of data.
+
+A Base64 character is nothing more than an alphabet letter (or the `+`, `/` and `=` signs). Check out the full [conversion table](https://en.wikipedia.org/wiki/Base64#Base64_table).
+
+When shipping data accross a network, it's never done by just sending over the binary data. This is because it is uncertain that it will work. Some protocols might think you inserted a special character combination (like a line ending) instead. For this reason, Base64 encoding is used.
+
+If you're wondering why 64 and not another number, it's because those 64 characters are almost always present, and your data will be transferred without corruptions.
```
Comment (nem035) [inline glossary/general/base64-encoding.md:9]: I tried to explain it a bit more in plain English, what do you think?

```suggestion
# Base64 Encoding

In computers, files are commonly stored in binary format (strings of 0's and 1's) but data on the internet is usually transferred as text.

Base64 represents a way to convert binary data into text using 64 [ASCII](https://en.wikipedia.org/wiki/ASCII) characters.

It is predominantly used on the web for encoding files such as images, or sounds into textual formats native to the web such as HTML and CSS.

A Base64 character can an English alphabet letter or the `+`, `/` or `=` sign. Check out the full [conversion table](https://en.wikipedia.org/wiki/Base64#Base64_table).
```

**Comment by enki-hq** [comment]
Rule insight:footnotes failed for files: 
  - javascript/browser-apis/reading-writing/filereader-readasdataurl.md
